From c6662dbedc210f337b4c4b497b48313d0680c182 Mon Sep 17 00:00:00 2001
From: Danesh Mondegarian <daneshm90@gmail.com>
Date: Sat, 6 Apr 2013 12:26:31 -0400
Subject: [PATCH] Lock screen weather : Switch to new yahoo api

Use YQL since where.yahooapis.com seems to have been shutdown

Change-Id: I53c981383ea6836466ef854b1e2f1db17493e812
---
 .../android/internal/util/weather/YahooPlaceFinder.java  |   14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/core/java/com/android/internal/util/weather/YahooPlaceFinder.java b/core/java/com/android/internal/util/weather/YahooPlaceFinder.java
index fb8efe4..fbc4017 100644
--- a/core/java/com/android/internal/util/weather/YahooPlaceFinder.java
+++ b/core/java/com/android/internal/util/weather/YahooPlaceFinder.java
@@ -17,25 +17,27 @@
 package com.android.internal.util.weather;
 
 import android.content.Context;
+import android.net.Uri;
 
 public class YahooPlaceFinder {
 
-    private static final String YAHOO_API_BASE_REV_URL = "http://where.yahooapis.com/geocode?appid=jYkTZp64&q=%1$s,+%2$s&gflags=R";
-    private static final String YAHOO_API_BASE_URL = "http://where.yahooapis.com/geocode?appid=jYkTZp64&q=%1$s";
+    private static final String YAHOO_API_BASE_URL = "http://query.yahooapis.com/v1/public/yql?q=" +
+            Uri.encode("select woeid from geo.placefinder where text =");
 
     public static String reverseGeoCode(Context c, double latitude, double longitude) {
 
-        String url = String.format(YAHOO_API_BASE_REV_URL, String.valueOf(latitude),
-                String.valueOf(longitude));
+        String formattedCoordinates = String.format("\"%s %s\" and gflags=\"R\"",
+                String.valueOf(latitude), String.valueOf(longitude));
+        String url = YAHOO_API_BASE_URL + Uri.encode(formattedCoordinates);
         String response = new HttpRetriever().retrieve(url);
         return new WeatherXmlParser(c).parsePlaceFinderResponse(response);
 
     }
 
     public static String GeoCode(Context c, String location) {
-        String url = String.format(YAHOO_API_BASE_URL, location).replace(' ', '+');
+        String url = String.format(YAHOO_API_BASE_URL + Uri.encode(String.format("\"%s\"",location));
         String response = new HttpRetriever().retrieve(url);
         return new WeatherXmlParser(c).parsePlaceFinderResponse(response);
     }
 
-}
+}
\ No newline at end of file
-- 
1.7.10.4

